scripts/src/java/org/oppia/android/scripts/coverage/RunCoverage.kt

```diff
@@ -0,0 +1,153 @@
+package org.oppia.android.scripts.coverage
+
+import org.oppia.android.scripts.common.BazelClient
+import org.oppia.android.scripts.common.CommandExecutor
+import org.oppia.android.scripts.common.CommandExecutorImpl
+import org.oppia.android.scripts.common.ScriptBackgroundCoroutineDispatcher
+import org.oppia.android.scripts.proto.TestFileExemptions
+import java.io.File
+import java.io.FileInputStream
+import java.util.concurrent.TimeUnit
+
+/**
+ * Entry point function for running coverage analysis for a source file.
+ *
+ * Usage:
+ *   bazel run //scripts:run_coverage_for_test_target -- <path_to_root> <relative_path_to_file>
+ *
+ * Arguments:
+ * - path_to_root: directory path to the root of the Oppia Android repository.
+ * - relative_path_to_file: the relative path to the file to analyse coverage
+ *
+ * Example:
+ *     bazel run //scripts:run_coverage -- $(pwd)
+ *     utility/src/main/java/org/oppia/android/util/parser/math/MathModel.kt
+ * Example with custom process timeout:
+ *     bazel run //scripts:run_coverage -- $(pwd)
+ *     utility/src/main/java/org/oppia/android/util/parser/math/MathModel.kt processTimeout=10
+ *
+ */
+fun main(vararg args: String) {
```

Comment:
With the introduction of the entry point in RunCoverage, may be I can remove that in RunCoverageForTestTarget https://github.com/oppia/oppia-android/pull/5423/files#diff-04f7dc1ca7f9e9b2f3eac2fe9e9de7841146490f96e68f9059223c8c85588bfaR28 as that will no longer be used.

Reply:
Why not update ``RunCoverageForTestTarget`` instead of creating a new file? I'm a bit confused about the implementation order, at the moment. Why introduce ``RunCoverageForTestTarget`` at all?

Also, perhaps 'ComputeCoverageForSourceFile' might be a clearer name for this script--WDYT?

Reply:
Yeah! actually the `RunCoverageForTestTarget` script now just acts as an intermediary for passing data and nothing useful. Initially, I followed it to stick with what we decided, ie. to have '3 scripts' when we altered our initial plan. Originally, we planned to provide test target inputs and obtain coverage reports. However, we later decided it would be more user-friendly to use filename inputs instead.

With this change, we specified having three separate scripts:
1. Running coverage analysis.
2. Running a single target.
3. Running coverage for a specific file.

I viewed this as a separation of concerns and proceeded accordingly. However, the `bazelClient` already handles running coverage for a single test target, making this script redundant—it's now just an intermediary and nothing more.

Would you prefer I remove this script and consolidate it into two utilities instead: **RunCoverage** and **CoverageRunner**? We could possibly introduce a **CoverageReporter** later, perhaps with **PR 1.5** for generating reports.

Reply:
> Also, perhaps 'ComputeCoverageForSourceFile' might be a clearer name for this script--WDYT?

Uh... just few thoughts on why I initially named it **runCoverage**, I wanted to keep it simple and easy for developers to remember. So, I thought we could keep it abstract, as developers would follow the wiki instructions and not everyone (especially those coming from gradle using bazel just for coverages) would delve into whether it's running for source files or test targets.

However, naming it runCoverage might cause confusion for those who are working with bazel and know how it works, as they might expect it to run test targets so having a proper naming would work. But this wouldn't mean much as over time, developers would get used to the command, so a descriptive command could work too.

I've been used to calling it runCoverage for a while now, but if renaming it makes things clearer, I can refactor it while simplifying the scripts into one.

Reply:
Yeah Ben, I think we can remove **RunCoverageForTestTarget** and I proceeded to do that and updated this PR too.
[[PR 1.1 commit](https://github.com/oppia/oppia-android/pull/5423/commits/c834aa77f3e6e71b3f15dcf7d606a2f889b4271b)]

Reply:
SGTM--this comment is addressed from my perspective but technically you're the one that opened it, so please resolve it once you're happy with things.

Reply:
Okay Done!